client/components/data/query-site-purchases/index.jsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isFetchingSitePurchases } from 'state/purchases/selectors';
+import { fetchSitePurchases } from 'state/purchases/actions';
+
+class QuerySitePurchases extends Component {
+	constructor( props ) {
+		super( props );
+		this.requestSitePurchases = this.requestSitePurchases.bind( this );
+	}
+
+	requestSitePurchases( props = this.props ) {
+		this.props.fetchSitePurchases( props.siteId );
+	}
+
+	componentWillMount() {
+		this.requestSitePurchases();
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.fetchingSitePurchases ||
+			! nextProps.siteId ||
+			( this.props.siteId === nextProps.siteId ) ) {
+			return;
+		}
+		this.requestSitePurchases( nextProps );
```

Comment:
What's the reason for using a condition to exit the method whereas we could save one line with:

```javascript
if ( ! nextProps.fetchingSitePurchases && nextProps.siteId &&
  ( this.props.siteId !== nextProps.siteId ) ) {
  this.requestSitePurchases( nextProps );
}
```